qiskit/quantum_info/operators/symplectic/sparse_pauli_op.py

```diff
@@ -0,0 +1,564 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+N-Qubit Sparse Pauli Operator class.
+"""
+
+from numbers import Number
+import numpy as np
+
+from qiskit.exceptions import QiskitError
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+from qiskit.quantum_info.operators.operator import Operator
+from qiskit.quantum_info.operators.symplectic.pauli_table import PauliTable
+from qiskit.quantum_info.operators.symplectic.pauli_utils import pauli_basis
+from qiskit.quantum_info.operators.custom_iterator import CustomIterator
+
+
+class SparsePauliOp(BaseOperator):
+    """Sparse N-qubit operator in a Pauli basis representation.
+
+    This is a sparse representation of an N-qubit matrix
+    :class:`~qiskit.quantum_info.Operator` in terms of N-qubit
+    :class:`~qiskit.quantum_info.PauliTable` and complex coefficients.
+
+    It can be used for performing operator arithmetic for hundred of qubits
+    if the number of non-zero Pauli basis terms is sufficiently small.
+
+    The Pauli basis components are stored as a
+    :class:`~qiskit.quantum_info.PauliTable` object and can be accessed
+    using the :attr:`~SparsePauliOp.pauli` attribute. The coefficients
+    are stored as a complex Numpy array vector and can be accessed using
+    the :attr:`~SparsePauliOp.coeffs` attribute.
+    """
+
+    def __init__(self, data, coeffs=None):
+        """Initialize an operator object."""
+        if isinstance(data, SparsePauliOp):
```

Comment:
What the purpose of this part of the constructor? Is it to copy somehow? Or is it supposed to be checking that data is a PauliTable?

Reply:
Its a shallow copy constructor